src/Solution/ParticleTracker/MethodDis.f90

```diff
@@ -182,6 +198,23 @@ subroutine load_particle(this, cell, particle)
       idiag = this%fmi%dis%con%ia(cell%defn%icell)
       ipos = idiag + inbr
       ic = dis%con%ja(ipos)
+
+      if (ic == particle%icp) then
+        ! terminate in the previous cell.
+        ! got here by falling through the
+        ! bottom of a well.. TODO: check
+        ! if entry/exit face is top/bot?
+        particle%advancing = .false.
+        particle%idomain(2) = particle%icp
+        particle%istatus = 2
+        particle%izone = particle%izp
+        call this%save(particle, reason=3)
+        return
+      else
+        particle%icp = particle%idomain(2)
+        particle%izp = particle%izone
+      end if
+
```

Comment:
as mentioned above, maybe we can refine this so as not to prevent legitimate reentires